### ISSUE TITLE
RasterizeState.FillMode state isn't tracked

### ISSUE THREAD

**Issue by RUSshy** [tlgkccampbell/ultraviolet#104]
With this code:

```c#

    [EcsInject]
    public class DebugRenderer : IEcsRunSystem, IEcsInitSystem
    {
        public EcsWorld world;
        public EcsFilter<CModel, CTransform> filter;

        public Assets assets;
        public PerspectiveCamera _camera;
        
        public Mesh mesh;
        public Effect effect;
        private RasterizerState _wireframeState;
        
        public void initialize()
        {
            effect = assets.manager.Load<Effect>("effects/debug.vert");
            mesh = TestSystem.genCube(1);
            
            _wireframeState = RasterizerState.Create();
            _wireframeState.CullMode = CullMode.None;
            _wireframeState.FillMode = FillMode.Wireframe;
        }

        public void destroy()
        {
        }

        public void run(float dt)
        {
            var device = Core.context.GetGraphics();
            var originalState = device.GetRasterizerState();

            
            device.SetBlendState(BlendState.AlphaBlend);
            device.SetRasterizerState(_wireframeState);

            foreach (var id in filter)
            {
                var cModel = filter.components1[id];
                var cTransform = filter.components2[id];
                
                // todo: change size to match bounding box

                effect.Parameters["u_proj"].SetValue(_camera.projection);
                effect.Parameters["u_view"].SetValue(_camera.view);
                effect.Parameters["u_world"].SetValue(cTransform.matrix);

                effect.CurrentTechnique.Passes[0].Apply();
                mesh.render(device, PrimitiveType.LineList);
            }
            
            device.SetRasterizerState(originalState);
        }
    }
```

I store old state wich is FillMode.Solid, but for debug rendering i render some stuff with Wireframe

So i set the state with Wirefram, then i restore the old Solid state

But then everything else is rendered as wireframe

**Comment by RUSshy** [comment]
![dotnet_2018-11-11_19-40-02](https://user-images.githubusercontent.com/18348637/48316887-9b460c80-e5e9-11e8-876e-d65573f52b53.png)

**Comment by tlgkccampbell** [comment]
Fixed in aca71860aec711c16a4feab33a0111b1e22711ae
